src/main/java/bio/terra/workspace/service/job/JobService.java

```diff
@@ -115,9 +117,18 @@ public JobBuilder newJob(
   // submit a new job to stairway
   // protected method intended to be called only from JobBuilder
   protected String submit(
-      Class<? extends Flight> flightClass, FlightMap parameterMap, String jobId) {
+      Class<? extends Flight> flightClass,
+      FlightMap parameterMap,
+      String jobId,
+      boolean duplicateFlightOk) {
     try {
       stairway.submit(jobId, flightClass, parameterMap);
+    } catch (DuplicateFlightIdSubmittedException ex) {
+      if (duplicateFlightOk) {
+        return jobId;
```

Comment:
probably worth some explanatory comment here or elsewhere on why this is nice behavior to have.
also, the catch ordering matters and that's not immediate obvious, so worth a warning comment.

Reply:
sg. I think it'll be worth adding a slightly longer explanation in the JobBuilder, but I'm less concerned about inside a protected method.